core/integration/trg/plugin.go

```diff
@@ -853,6 +866,10 @@ func (p *Plugin) CallStack(data interface{}) (stack map[string]interface{}) {
 			return
 		}
 
+		timeout := callable.AcquireTimeout(TRG_START_TIMEOUT, varStack, "RunStart", envId)
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+
```

Comment:
why are you creating the context with timeout here and not next to the place where it is needed? (line 889)

Reply:
I just did it in the same spot as in `ddsched`. Shouldn't change much either way, as between here and the call we only have declarations and a Kafka event.